src/id/onepieceberwarna/src/eu/kanade/tachiyomi/extension/id/onepieceberwarna/OnePieceBerwarna.kt

```diff
@@ -0,0 +1,143 @@
+
+        run loop@{
+            sections.forEach { element ->
+                val dataID = element.attr("data-id")
+                if (dataID == "49afe94") {
+                    // break on non-manga section
+                    return@loop
+                }
+
+                element.select(".elementor-col-100:not(:has(div[data-element_type=icon.default])) h1")
+                    .first()?.let {
+                        saga = it.text()
+                    }
+
+                element.select(".elementor-col-50 h1").first()?.let {
+                    volume = it.text()
+                }
+
+                element.select(".elementor-widget-container > img").let {
+                    if (it.hasAttr("data-src")) {
+                        image = it.attr("data-src")
+                        list.add(
+                            SManga.create().apply {
+                                title = "$saga $volume"
+                                thumbnail_url = image
+                                url = dataID // just for identifier
+                            },
+                        )
+                    }
+                }
+            }
+        }
+        return MangasPage(list, false)
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga = throw UnsupportedOperationException("Not used")
+
+    override fun popularMangaNextPageSelector() = throw UnsupportedOperationException("Not used")
+
+    override fun mangaDetailsRequest(manga: SManga) = GET(baseUrl, headers)
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        document.select("section:first-child span").forEach {
+            val title = it.text()
+            if (title.contains("Credit")) {
+                author = title.substringAfter("to")
+            }
+            if (title.contains("Colored")) {
+                artist = title.substringAfter("by")
+            }
+        }
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> = Single.create {
+        val response = client.newCall(GET(baseUrl, headers)).execute()
+        val doc = response.asJsoup()
+
+        it.onSuccess(
+            // get section base on data-id stored in manga.url
+            doc.select("section[data-id=${manga.url}] .elementor-text-editor strong > a")
+                .map { element ->
+                    SChapter.create().apply {
+                        name = element.text()
+                        setUrlWithoutDomain(element.attr("href"))
+                    }
+                },
+        )
+    }.toObservable()
+
+    override fun chapterListSelector(): String = throw UnsupportedOperationException("Not used")
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException("Not used")
+
+    override fun pageListParse(document: Document): List<Page> {
+        return document.select(".entry-content img")
```

Comment:
```suggestion
        return document.select(".entry-content img:not(a > img):is([data-src])")
```
can filter non manga img entirely with css selector instead or something similar

Reply:
`rel` adanya di `a` kan ya itu sama exclude

Reply:
ada beberapa ads yang nggak masuk ke `a > img` tapi semua ads pake parent `rel` jadi saya pake `rel`

Reply:
kalo ini di ilangin `>` masuk tak

Reply:
oh iya, bisa kalo dihilangin `>` nya